### PR TITLE
Migrate Switzerland holidays to holiday groups, localization

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Migrate Switzerland holidays to `holiday_groups` standard, localization (de, en_US, fr, it, uk).

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/switzerland.py [hunk 1]
```diff
@@ -9,227 +9,267 @@
...
-        if self.subdiv == "GE":
-            self[date(year, DEC, 31)] = "Wiederherstellung der Republik"
+            # National Day.
+            self._add_holiday(tr("Nationalfeiertag"), AUG, 1)
+
+        # Christmas Day.
+        self._add_christmas_day(tr("Weihnachten"))
+
+    def _add_subdiv_holidays(self):
+        if self.subdiv and self.subdiv != "VS":
```
Comment (arkid15r) [inline holidays/countries/switzerland.py:81]: Consider adding
```
if not self.subdiv:
    return None
```

File: holidays/countries/switzerland.py [hunk 2]
```diff
@@ -9,227 +9,267 @@
...
+            # St. Stephen's Day.
+            self._add_christmas_day_two(tr("Stephanstag"))
+
+        super()._add_subdiv_holidays()
+
+    def _add_subdiv_ag_holidays(self):
+        # Berchtold's Day.
+        self._add_new_years_day_two(tr("Berchtoldstag"))
+
+    def _add_subdiv_ar_holidays(self):
```
Comment (arkid15r) [inline holidays/countries/switzerland.py:101]: Not sure if we want to keep dummy subdiv methods.

File: holidays/countries/switzerland.py [hunk 3]
```diff
@@ -9,227 +9,267 @@
...
+        if self._year >= 1835:
+            dt = _get_nth_weekday_of_month(1, THU, APR, self._year)
+            if dt == self._easter_sunday + td(days=-3):
+                dt += td(days=+7)
+            # Näfels Ride.
+            self._add_holiday(tr("Näfelser Fahrt"), dt)
+
+        self._add_all_saints_day(tr("Allerheiligen"))
+
+    def _add_subdiv_gr_holidays(self):
```
Comment (arkid15r) [inline holidays/countries/switzerland.py:154]: This one too - I suggest removing them.

File: holidays/locale/en_US/LC_MESSAGES/CH.po [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+msgstr "Immaculate Conception"
+
+#. Labour Day.
+#: holidays/countries/switzerland.py:119 holidays/countries/switzerland.py:122
+#: holidays/countries/switzerland.py:159 holidays/countries/switzerland.py:179
+#: holidays/countries/switzerland.py:207 holidays/countries/switzerland.py:220
+#: holidays/countries/switzerland.py:224 holidays/countries/switzerland.py:229
+#: holidays/countries/switzerland.py:272
+msgid "Tag der Arbeit"
+msgstr "Labour Day"
```
Comment (arkid15r) [inline holidays/locale/en_US/LC_MESSAGES/CH.po:123]:
```suggestion
msgstr "Labor Day"
```

File: holidays/locale/en_US/LC_MESSAGES/CH.po [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+#. Jeûne genevois.
+#: holidays/countries/switzerland.py:134
+msgid "Jeûne genevois"
+msgstr "Jeûne genevois"
```
Comment (arkid15r) [inline holidays/locale/en_US/LC_MESSAGES/CH.po:128]: Not really sure:
```suggestion
msgstr "Genevan Fast"
```

At least adds some context.

File: holidays/locale/en_US/LC_MESSAGES/CH.po [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+#. Restoration Day.
+#: holidays/countries/switzerland.py:139
+msgid "Wiederherstellung der Republik"
+msgstr "Restoration Day"
+
+#. Näfels Ride.
+#: holidays/countries/switzerland.py:150
+msgid "Näfelser Fahrt"
+msgstr "Näfels Ride"
```
Comment (arkid15r) [inline holidays/locale/en_US/LC_MESSAGES/CH.po:138]:
```suggestion
msgstr "Battle of Naefels Victory Day"
```

**Review by arkid15r — COMMENTED** [review]
Tremendous work 👍

File: holidays/countries/switzerland.py (hunk 2 above)
Comment (KJhellico) [reply holidays/countries/switzerland.py:101]: I left them just to show that we didn't accidentally miss these subdivs.

File: holidays/locale/en_US/LC_MESSAGES/CH.po (hunk 5 above)
Comment (KJhellico) [reply holidays/locale/en_US/LC_MESSAGES/CH.po:128]: I was guided by [English wiki](https://en.wikipedia.org/wiki/Je%C3%BBne_genevois), although I didn't really like it myself.

**Pull request merged**